spec/core/config_mixin_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "logstash/config/mixin"
+require "logstash/filters/base"
+
+describe LogStash::Config::Mixin do
+  context "when validating :bytes" do
+    it "should validate :bytes" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes
+      end.new({"size_file" => "10Kib"})
+      expect(conf_class.size_file).to eq(10240)
+    end
+
+    it "should allow setting valid default" do
+      conf_class = Class.new(LogStash::Filters::Base) do
+        include LogStash::Config::Mixin
+        config_name "test"
+        milestone 1
+        config :size_file, :validate => :bytes, :default => "10Kib"
+      end.new({})
+      expect(conf_class.size_file).to eq(10240)
+    end
```

Comment:
I have looked at the `filesize`'s gem code and the unit check is not case sensitive, we should have a test to guard this behavior.